contrib/epee/src/byte_slice.cpp

```diff
@@ -217,4 +232,29 @@ namespace epee
     portion_ = nullptr;
     return out;
   }
+
+  byte_buffer byte_buffer_resize(byte_buffer buf, const std::size_t length) noexcept
+  {
+    if (std::numeric_limits<std::size_t>::max() - sizeof(raw_byte_slice) < length)
+      return nullptr;
+
+    std::uint8_t* data = buf.get();
+    if (data != nullptr)
+      data -= sizeof(raw_byte_slice);
+
+    buf.release();
+    data = static_cast<std::uint8_t*>(std::realloc(data, sizeof(raw_byte_slice) + length));
```

Comment:
Memory leak.

Reply:
yeah crap, the release must be moved after the `nullptr` check.